### PR TITLE
fix: require selected fulfillment option to complete shipping step

### PR DESCRIPTION
**Pull request by oxaudo**
Hopefully addresses: https://artsyproduct.atlassian.net/browse/EMI-3266

  The issue was that we only considered the presence of `addressLine1`, `country`, and `name` as requirements for completing the step. This
  worked fine when the step only handled shipping addresses, but now that it's been expanded to "fulfillment details" (including delivery
  option selection), those rules have become insufficient.

  The step completion logic is encapsulated in `useCompleteFulfillmentDetailsData`, which is good for maintainability. However, I'm not
  entirely happy with how much business logic lives in the frontend. Ideally, we could move this logic to Metaphysics, but that refactor
  doesn't seem worth it at this point.

  This fix adds the requirement that BUY orders must have a `selectedFulfillmentOption` before the step can complete. I don't have high
  confidence that this won't break some edge case flows, but our test coverage is extensive and should catch most issues.

### CODE REVIEW

File: src/Apps/Order2/Routes/Checkout/Components/DeliveryOptionsStep/__tests__/Order2DeliveryOptionsForm.jest.tsx [hunk 1]
```diff
@@ -381,15 +381,14 @@ describe("Order2DeliveryOptionsForm", () => {
         name: "Continue to Payment",
       })
 
-      await userEvent.click(continueButton)
```
Comment (oxaudo) [inline src/Apps/Order2/Routes/Checkout/Components/DeliveryOptionsStep/__tests__/Order2DeliveryOptionsForm.jest.tsx:384]: Not sure how this test was working before unless it was actually testing the bug presense?

Comment (dariakoko) [reply src/Apps/Order2/Routes/Checkout/Components/DeliveryOptionsStep/__tests__/Order2DeliveryOptionsForm.jest.tsx:384]: I think we should keep `await userEvent.click(continueButton)`

It verifies that the disabled state prevents side effects, which is what we’re testing with:
1. `expect(mockSetOrderFulfillmentOption).not.toHaveBeenCalled()`
2. `expect(mockCheckoutContext.completeStep).not.toHaveBeenCalled()`

I would also keep `expect(continueButton).toBeDisabled()` that you added, since it validates the UI state, which is important too

The click on the disabled button tests behavior, while `toBeDisabled()` tests state and together they give us better coverage

File: src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts [hunk 2]
```diff
@@ -17,41 +16,58 @@ export const useCompleteFulfillmentDetailsData = (
...
+      isOffer,
+    }
+  }
+
+  /**
+   * For delivery addresses:
+   * - In BUY mode: requires both address data AND a valid fulfillment option to be selected.
+   *   If address exists but no fulfillment option is set, this indicates an error state
+   *   (e.g., missing postal code, unsupported region) and the step should remain active.
+   * - In OFFER mode: address data alone is sufficient since shipping is determined later (TBD).
```
Comment (dariakoko) [inline src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts:39]: Is "TBD" related to some outstanding work in this PR?

File: src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts [hunk 3]
```diff
@@ -17,41 +16,58 @@ export const useCompleteFulfillmentDetailsData = (
...
+   *   However, we still validate that basic required fields are present.
+   *
+   * For pickup: step is complete if name and phone number exist and PICKUP is selected.
+   */
+  const isComplete = ["addressLine1", "country", "name"].every(field => {
```
Comment (dariakoko) [inline src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts:42]: Do we need to include the phone number in this list as well?

File: src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts [hunk 4]
```diff
@@ -17,41 +16,58 @@ export const useCompleteFulfillmentDetailsData = (
...
+    return fulfillmentDetails[field as keyof typeof fulfillmentDetails] != null
+  })
+
+  if (!isComplete) {
+    return null
+  }
+
+  // In BUY mode fulfillment option has to be selected.
+  // If address exists but no option is selected, step is incomplete.
```
Comment (dariakoko) [inline src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts:51]: > fulfillment option has to be selected
In what cases the fulfillment option is not selected? Asking because we select it automatically upon landing on the page and I want to test the case when it's not selected
Sorry if I'm just not understanding it fully 🤔

File: src/Apps/Order2/Routes/Checkout/Components/DeliveryOptionsStep/__tests__/Order2DeliveryOptionsForm.jest.tsx (hunk 1 above)
Comment (oxaudo) [reply src/Apps/Order2/Routes/Checkout/Components/DeliveryOptionsStep/__tests__/Order2DeliveryOptionsForm.jest.tsx:384]: It was failing for me I assumed because `await userEvent.click(continueButton)` never resolves for disabled button. That is why I was wondering how it was working before...

File: src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts (hunk 2 above)
Comment (oxaudo) [reply src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts:39]: "TBD" is a valid shipping option for make offer case that gets assigned when we can not determine any other shipping options but proceed to payment. It's displayed as 'to be determined by seller' in UI.

This state:
Screenshot 2026-06-10 at 2 44 38 PM

This comment ment to explain why we are allowing the case for offer where `selectedFulfilmentOption` is null later in the code.
It is super confusing I agree - happy to chat in person to brainstorm how to make it better.

File: src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts (hunk 3 above)
Comment (oxaudo) [reply src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts:42]: Opps. Pickup case is actually handled above. And for that we do not check address presence at all. I was just reshuffling the comments and copied this pickup one to the wrong place. Cleaning up.

This section only applies to SHIP fulfilmentType as PICKUP handled above

File: src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts (hunk 4 above)
Comment (oxaudo) [reply src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts:51]: It is this case of 'incomplete shipping state' that I pointed at above. We need address to be selected and 'fulfillment_type' to be selected. But fulfillment option stays nil here as we don't really have UI expression for 'shipping TBD'. It's selected silently when we move to the next step.

Again. None of this is intuitive really but that is the way our orders setup now:(

Part of the confusion here is that we have fulfilmentType (ship/pickup) that must be selected always and always selected by default as SHIP. And then fulfilmentOption gets selected in the second part of now combined address/ship step.

File: src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts [hunk 5]
```diff
@@ -17,41 +16,56 @@ export const useCompleteFulfillmentDetailsData = (
...
-          city: fulfillmentDetails?.city || null,
-          region: fulfillmentDetails?.region || null,
-          postalCode: fulfillmentDetails?.postalCode || null,
-          country: orderData.fulfillmentDetails?.country || null,
-          phoneNumber:
-            orderData.fulfillmentDetails?.phoneNumber?.display || null,
-        },
-      }
+  // For pickup: step is complete if name and phone number exist and PICKUP is selected.
+  if (selectedType === "PICKUP") {
```
Comment (starsirius) [inline src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts:23]: It may not be supported that pickup is selected/persisted and still incomplete, but I guess it's probably good to handle incomplete data for pickup anyway?

Comment (oxaudo) [reply src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts:23]: Yea - thought about it but this part I has not changed just moved it around here. We can try to see what would happen if one tries to load pickup order with order phone is missing (think it's only possible in console - no UI to be in this state) but will verify. If an issue - suggest treating it as a separate ticket anyways.

File: src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts (hunk 2 above)
Comment (dariakoko) [reply src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts:39]: Ah wow I see. Thanks for explaining
I'm wondering how the user journey will look in this case - would they receive an email with the shipping method details and the final cost?

File: src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts (hunk 3 above)
Comment (dariakoko) [reply src/Apps/Order2/Routes/Checkout/Components/FulfillmentDetailsStep/useCompleteFulfillmentDetailsData.ts:42]: I didn't even notice that 😄  Good catch!
My question was bout the list of the data required to mark the fulfillment step as complete: `["addressLine1", "country", "name"]`. Since the phone number is required on the address details form, should we add it to the array as well? Like this: `["addressLine1", "country", "name", "phoneNumber"]`

**Review by starsirius — APPROVED** [review]
LGTM!!

**Review by dariakoko — APPROVED** [review]
